### PR TITLE
way better apparmor docs

### PR DESCRIPTION
**Pull request by jessfraz**
ping @icecrime wdyt?

### CODE REVIEW

**Comment by calavera** [comment]
:heart_eyes_cat:

**Comment by icecrime** [comment]
Woah thanks :+1:

**Comment by MHBauer** [comment]
renders kind of funky.

screen shot 2016-01-26 at 14 39 07

screen shot 2016-01-26 at 14 39 12

I think those are supposed to be more like subheadings, but they don't look quite right to me. Not sure what the change to be made is. I'll play with it.

Seems helpful and nicely done. I don't know anything about apparmor, but now I can pretend I do.

**Comment by MHBauer** [comment]
suggestion, but requires confirmation from others:

use:
```
### Using `dmesg`
```
instead of
```
**`dmesg`**
```
That way, it looks like a header, and yet you still have monospace for the command name.

screen shot 2016-01-26 at 14 47 15

Markdown inside of markdown, yeesh.

**Comment by jfrazelle** [comment]
ah nice will do

File: docs/security/apparmor.md [hunk 1]
```diff
@@ -67,9 +73,111 @@ explicitly specifies the default policy:
...
+
+AppArmor sends quite verbose messaging to `dmesg`. Usually an AppArmor line
+will look like the following:
+
+```
+[ 5442.864673] audit: type=1400 audit(1453830992.845:37): apparmor="ALLOWED" operation="open" profile="/usr/bin/docker" name="/home/jessie/docker/man/man1/docker-attach.1" pid=10923 comm="docker" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
+```
+
+In the above example, the you can see `profile=/usr/bin/docker`. This means the
+user has the `docker-engine` (Docker Engine Daemon) profile loaded. On version
```
Comment (thaJeztah) [inline docs/security/apparmor.md]: Perhaps we should change the last sentence into a "note"? i.e. for Ubuntu 12.04 ..

File: docs/security/apparmor.md [hunk 2]
```diff
@@ -67,9 +73,111 @@ explicitly specifies the default policy:
...
+37 processes are in enforce mode.
+   docker-default (6044)
+   ...
+   docker-default (31899)
+1 processes are in complain mode.
+   /usr/bin/docker (29756)
+0 processes are unconfined but have a profile defined.
+```
+
+In the above output you can tell I have the `docker-default` profile running on
```
Comment (thaJeztah) [inline docs/security/apparmor.md]: "In the above output you can tell that the ..."

File: docs/security/apparmor.md [hunk 3]
```diff
@@ -67,9 +73,111 @@ explicitly specifies the default policy:
...
+various container PIDs in `enforce` mode. This means AppArmor will actively
```
Comment (thaJeztah) [inline docs/security/apparmor.md]: s/in/is in/

File: docs/security/apparmor.md [hunk 4]
```diff
@@ -67,9 +73,111 @@ explicitly specifies the default policy:
...
+block and audit in `dmesg` anything outside the bounds of the `docker-default`
```
Comment (thaJeztah) [inline docs/security/apparmor.md]: does it "audit in `dmesg`" or, "show audit *logs* in `dmesg`"?

**Comment by thaJeztah** [comment]
Cool @jfrazelle, some nits, but overall looking good \o/

File: docs/security/apparmor.md (hunk 4 above)
Comment (MHBauer) [inline docs/security/apparmor.md]: I think `in dmesg` is usually interpreted as `in the output of dmesg`. Really it's in the kernel's message buffer, which is regular logs. I feel like any addition to this explanation gets more into the weeds than content. I'd take another opinion on how others read it.

**Comment by jfrazelle** [comment]
updated!

File: docs/security/apparmor.md [hunk 5]
```diff
@@ -16,10 +16,16 @@ operating system and its applications from security threats. To use it, a system
...
-itself also exists and is installed with the official *.deb* packages in
-`/etc/apparmor.d/docker` file.
-
+Docker automatically loads container profiles. The Docker binary installs
+a `docker-default` profile in the `/etc/apparmor.d/docker` file. This profile
+is used on containers, _not_ on the Docker Daemon.
+
+A profile for the Docker Engine Daemon exists but it is not currently installed 
+with the deb packages. If you are interested in the source for the Daemon
+profile, it is
```
Comment (vdemeester) [inline docs/security/apparmor.md]: Too much carriage return :sweat_smile: :scream_cat:

Comment (jfrazelle) [inline docs/security/apparmor.md]: I blame linux! lol

**Comment by thaJeztah** [comment]
LGTM, thanks!

ping @vdemeester

**Comment by vdemeester** [comment]
LGTM 🐯

**Pull request merged**